app/controllers/benefits/CompanyBenefitController.scala

```diff
@@ -113,20 +118,40 @@ class CompanyBenefitController @Inject()(
         }
       },
       success => {
-        success match {
-          case Some(NoIDontGetThisBenefit) =>
-            journeyCacheService.cache(DecisionChoice, NoIDontGetThisBenefit) map { _ =>
-              Redirect(controllers.benefits.routes.RemoveCompanyBenefitController.stopDate())
-            }
-          case Some(YesIGetThisBenefit) =>
-            journeyCacheService.cache(DecisionChoice, YesIGetThisBenefit) map { _ =>
-              Redirect(
-                controllers.routes.ExternalServiceRedirectController
-                  .auditInvalidateCacheAndRedirectService(TaiConstants.CompanyBenefitsIform)
-                  .url)
+        val benefitType = journeyCacheService.mandatoryJourneyValueAs(
+          EndCompanyBenefit_BenefitTypeKey,
+          x => x
+        )
```

Comment:
x => x doesn't do anything here, you can use 

```suggestion
        ) journeyCacheService.mandatoryJourneyValue(EndCompanyBenefit_BenefitTypeKey)
```

Reply:
I remember trying this, and it leads to type ambiguity (if I understand the error correctly).

Reply:
Try changing it to (note removing the As)

>     val benefitType = journeyCacheService.mandatoryJourneyValue(EndCompanyBenefit_BenefitTypeKey)